deploy/eck-integrations/values.yaml

```diff
@@ -0,0 +1,493 @@
+  ##    securityContext: {}
+  ##    rules: []
+  ##    nodeSelector: {}
+  ##    tolerations: []
+  ##    topologySpreadConstraints: []
+  ##    extraEnv: []
+  ##    extraVolumes: []
+  ##    extraVolumeMounts: []
+  ##    https://github.com/elastic/elastic-agent/blob/main/_meta/elastic-agent.yml
+  ##    agent.monitoring: {}
+  ##    https://www.elastic.co/guide/en/fleet/current/providers.html
+  ##    providers: {}
+    clusterWide:
+      ## required by the built-in kubernetes integration
+      mode: deployment
+      annotations:
+        eck.k8s.elastic.co/license: basic
+      resources:
+        limits:
+          memory: 700Mi
+        requests:
+          cpu: 100m
+          memory: 400Mi
+      nodeSelector:
+        kubernetes.io/os: linux
+      extraVolumes:
+        - name: agent-data
+          emptyDir: {}
+      extraVolumeMounts:
+        - name: agent-data
+          mountPath: /usr/share/elastic-agent/state
+      extraEnvs:
+      - name: ELASTIC_NETINFO
+        value: "false"
+      agent:
+        monitoring:
+          namespace: default
+          use_output: default
+          enabled: true
+          logs: true
+          metrics: true
+      providers:
+        kubernetes_leaderelection:
+          enabled: true
+        kubernetes:
+          node: ${NODE_NAME}
+          scope: cluster
+    perNode:
+      ## required by the built-in kubernetes and cloud-defend integrations
+      mode: daemonset
+      annotations:
+        eck.k8s.elastic.co/license: basic
+      resources:
+        limits:
+          memory: 700Mi
+        requests:
+          cpu: 100m
+          memory: 400Mi
+      securityContext:
+        runAsUser: 0
+      nodeSelector:
+        kubernetes.io/os: linux
+      extraEnvs:
+        - name: ELASTIC_NETINFO
+          value: "false"
+      agent:
+        monitoring:
+          namespace: default
+          use_output: default
+          enabled: true
+          logs: true
+          metrics: true
+      providers:
+        kubernetes_leaderelection:
+          enabled: false
+        kubernetes:
+          node: ${NODE_NAME}
+          scope: node
+    ksmSharded:
+      ## required by the built-in kubernetes integration
+      mode: statefulset
+      annotations:
+        eck.k8s.elastic.co/license: basic
+      resources:
+        limits:
+          memory: 700Mi
+        requests:
+          cpu: 100m
+          memory: 400Mi
+      extraVolumes:
+        - name: agent-data
+          emptyDir: {}
+      extraVolumeMounts:
+        - name: agent-data
+          mountPath: /usr/share/elastic-agent/state
+      nodeSelector:
+        kubernetes.io/os: linux
+      extraEnvs:
+        - name: ELASTIC_NETINFO
+          value: "false"
+      agent:
+        monitoring:
+          namespace: default
+          use_output: default
+          enabled: true
+          logs: true
+          metrics: true
+      providers:
+        kubernetes_leaderelection:
+          enabled: false
```

Comment:
This works to disable leader election properly via the agent CRD's config section?  Does this work to override the initial configurations for Fleet managed agents as well?

If so it is significantly easier than the hoops users are jumping through to do this without ECK today, see https://www.elastic.co/guide/en/fleet/current/advanced-kubernetes-managed-by-fleet.html.

Reply:
essentially via configRef which a secret that the chart populates accordingly and then the operator parses it and constructs properly the agent.yml but I think we mean the same here 🙂 I am not fully aware of how something similar could like for fleet-managed agents but definitely we can have a look

Reply:
What I hope happens is the same mechanism overrides the initial configuration of the agent to seed it with the providers disabled, and then Fleet takes over from that starting base configuration.

The providers can't be configured from Fleet right now so this is a pain point.

Reply:
Also, with the Helm chart being able to create both a Deployment and DaemonSet I don't think we need leaderelection at all anymore so it should be disabled everywhere by default (pending a closer review of the configurations to make sure we did this right).

The only reason leader election exists is so you can just deploy a DaemonSet and emulate having a dedicated deployment to simplify the configuration needed (AFAIK).